MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -2,11 +2,196 @@ import UIKit
+                                 question: model.text,
+                                 questionNumber: "\(currentQuestionIndex+1)/\(questions.count)")
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+                       
+        if isCorrect == true {
+            imageView.layer.borderColor = UIColor(named: "YP Green")?.cgColor
+            correctAnswer += 1
+        } else { imageView.layer.borderColor = UIColor(named: "YP Red")?.cgColor
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.showNextQuestionOrResults()
+        }
+                   
+
+        
+    }
+    
+    private func showNextQuestionOrResults() {
+      if currentQuestionIndex == questions.count - 1 { // - 1 потому что индекс начинается с 0, а длинна массива — с 1
+        // показать результат квиза
+          let text = "Ваш раунд результат: \(correctAnswer) из 10"
+          let viewModel = QuizResultsViewModel (title: "Этот раунд окончен", text: text, buttonText: "Сыграть еще раз")
+          show(quiz: viewModel)
+          
+      } else {
+        currentQuestionIndex += 1 // увеличиваем индекс текущего урока на 1; таким образом мы сможем получить следующий урок
+        // показать следующий вопрос
+          imageView.layer.masksToBounds = true
+          imageView.layer.borderWidth = 0
+          allowAnswer = true
+          let nextQuestion = questions[currentQuestionIndex]
+          let viewModel = convert(model: nextQuestion)
+          show(quiz: viewModel)
+          
+      }
+    }
+    
+    private func show(quiz result: QuizResultsViewModel) {
+        let alert = UIAlertController(
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert)
+            
+        let action = UIAlertAction(
+            title: result.buttonText,
+            style: .default) {_ in
+                self.currentQuestionIndex = 0
+                self.correctAnswer = 0
+                self.imageView.layer.masksToBounds = true
+                self.imageView.layer.borderWidth = 0
+                self.allowAnswer = true
+                let firstQuestion = questions[self.currentQuestionIndex]
+                let viewModel = self.convert(model: firstQuestion)
+                self.show(quiz: viewModel)
+                
+            }
+        alert.addAction(action)
+        
+        self.present(alert,animated: true, completion: nil)
+    }
+    
+    @IBAction private func nuButtoneClicked(_ sender: UIButton) {
+        if allowAnswer == true {
+            let currentQuestion = questions[currentQuestionIndex]
+            let givenAnswer = false
+            showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+            allowAnswer = false
+        }
```

Comment:
Если ты хочешь избежать выполнения некоторого кода, то лучше подходит guard 

```suggestion
        guard allowAnswer else { return }
        
        let currentQuestion = questions[currentQuestionIndex]
        let givenAnswer = false
        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
        allowAnswer = false
        
```